packages/community/src/llm/bedrock/base.ts

```diff
@@ -194,9 +197,59 @@ class AnthropicProvider extends Provider {
   }
 }
 
+class MetaProvider extends Provider {
+  getResultFromResponse(
+    response: Record<string, any>,
+  ): MetaNoneStreamingResponse {
+    return JSON.parse(toUtf8(response.body));
+  }
+
+  getTextFromResponse(response: Record<string, any>): string {
+    const result = this.getResultFromResponse(response);
+    return result.generation;
+  }
+
+  getTextFromStreamResponse(response: Record<string, any>): string {
+    const event: StreamEvent | undefined = response.chunk?.bytes
+      ? JSON.parse(toUtf8(response.chunk?.bytes))
+      : undefined;
+
+    if (event?.type === "content_block_delta") return event.delta.text;
+    return "";
+  }
+
+  getRequestBody<T extends ChatMessage>(
+    metadata: LLMMetadata,
+    messages: T[],
+  ): InvokeModelCommandInput | InvokeModelWithResponseStreamCommandInput {
+    let promptFunction: (messages: ChatMessage[]) => string;
+
+    if (metadata.model.startsWith("meta.llama3")) {
+      promptFunction = mapChatMessagesToMetaLlama3Messages;
+    } else if (metadata.model.startsWith("meta.llama2")) {
+      promptFunction = mapChatMessagesToMetaLlama2Messages;
+    } else {
+      throw new Error(`Metal model ${metadata.model} is not supported`);
```

Comment:
I think "Metal" here is supposed to be "Meta"

Reply:
Ops... Typo fixed